example/dns.go

```diff
@@ -0,0 +1,59 @@
+package example
+
+import (
+	"fmt"
+	"github.com/miekg/dns"
+)
+
+func (e *exampleSolver) handleDNSRequest(w dns.ResponseWriter, req *dns.Msg) {
+	msg := new(dns.Msg)
+	msg.SetReply(req)
+	switch req.Opcode {
+	case dns.OpcodeQuery:
+		for _, q := range msg.Question {
+			switch q.Qtype {
+			case dns.TypeA:
+				rr, err := dns.NewRR(fmt.Sprintf("%s 5 IN A 127.0.0.1", q.Name))
+				if err != nil {
+					msg.SetRcode(req, dns.RcodeNameError)
+				} else {
+					msg.Answer = append(msg.Answer, rr)
+				}
```

Comment:
nit: add breaks

Reply:
Refactored this away with the function break out